ocaml/xapi/cli_operations.ml

```diff
@@ -208,14 +208,19 @@ let host_license_of_r host_r editions =
     else
       0.
   in
-  let edition = host_r.API.host_edition in
-  let edition_short = List.hd
-      (List.filter_map (fun (a, (_, b, _)) -> if a = edition then Some b else None) editions) in
+  let edition_str = host_r.API.host_edition in
+  let edition_short =
+    editions
+    |> List.filter_map
+      V6_interface.(fun ed -> if ed.ed_title = edition_str
+                     then Some ed.ed_code
+                     else None)
+    |> List.hd in
```

Comment:
We should really get rid of this List.hd...

Reply:
Replace it with a safer List.hd using options?

Reply:
It's usually better to use a match to get the first element of a list, and explicitly handle the `[]` case. The exception raised by `List.hd` in case the list is empty is often not what you want.

Reply:
As the `edition_short` value is used in the host license record, should we handle a failure with `free`, `""`, or by raising an error?